### PR TITLE
Feat: Add tokens

### PR DESCRIPTION
**Pull request by RyRy79261**
## Description 

This PR adds a hook that allows adding veMENTO tokens to the user's wallet. Added this add token functionality to the ClaimConfirmation Component on the claim page, so users can do this after claiming and locking their MENTO

### Other Changes

Updated 'Check another wallet' button to logout of the session, which prompted some error handling to tell wagmi to to fetch claim status or gas when not connected. 

Also cleaned up gas estimation logic left over in useClaimMento

### CODE REVIEW

File: src/hooks/use-add-tokens.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      params: {
+        type: "ERC20", // Initially only supports ERC20, but eventually more!
+        options: {
+          address: mento.addresses[chainId].Locking, // The address that the token is at.
+          symbol: "veMENTO", // A ticker symbol or shorthand, up to 5 chars.
+          decimals: 18, // The number of decimals in the token
+        },
+      },
+    });
+    console.log(`veMento token added: ${wasAdded}`);
```
Comment (chapati23) [inline src/hooks/use-add-tokens.tsx:42]:
```suggestion
```

File: src/hooks/use-add-tokens.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      params: {
+        type: "ERC20", // Initially only supports ERC20, but eventually more!
+        options: {
+          address: mento.addresses[chainId].MentoToken, // The address that the token is at.
+          symbol: "MENTO", // A ticker symbol or shorthand, up to 5 chars.
+          decimals: 18, // The number of decimals in the token
+        },
+      },
+    });
+    console.log(`Mento token added: ${wasAdded}`);
```
Comment (chapati23) [inline src/hooks/use-add-tokens.tsx:27]:
```suggestion
```

would do the same as we discussed in gov UI here, no console.log but instead log potential errors to sentry

File: src/app/allocation/page.tsx [hunk 3]
```diff
@@ -101,6 +103,14 @@ const HasKYC = () => {
       <Button color="blue" href={"/claim"}>
         Claim Your MENTO
       </Button>
+      <div>
+        <Button onClick={addMento} color={"blush"}>
+          Add MENTO token to wallet
```
Comment (chapati23) [inline src/app/allocation/page.tsx:108]:
```suggestion
          Add the MENTO token to your wallet
```

File: src/app/allocation/page.tsx [hunk 4]
```diff
@@ -101,6 +103,14 @@ const HasKYC = () => {
...
+        </Button>
+        <Button onClick={addVeMento} color={"blush"}>
+          Add veMENTO token to wallet
```
Comment (chapati23) [inline src/app/allocation/page.tsx:111]:
```suggestion
          Add the veMENTO token to your wallet
```

**Review by chapati23 — APPROVED** [review]
just a wording suggestion, but overall lgtm

**Review by bayological — APPROVED** [review]

**Comment by denviljclarke** [comment]
@bayological @chapati23. Made a few small changes to this. I think the layout looks better now, I also think we don't need to add MENTO here, only veMENTO.. but tell me what you think.

Before:
Screenshot 2024-06-25 at 3 32 49 PM
Screenshot 2024-06-25 at 3 33 11 PM

After:

Screenshot 2024-06-25 at 3 30 35 PM
Screenshot 2024-06-25 at 3 30 56 PM

**Review by chapati23 — APPROVED** [review]
lgtm

**Review by baroooo — APPROVED** [review]

**Pull request merged**